scripts/release/generate_history_notes.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import datetime
+import fileinput
+import json
+import re
+import requests
+
+
+base_url = 'https://api.github.com/repos/azure/azure-cli'
+release_head_url = '{}/git/ref/heads/release'.format(base_url)
+dev_commits_url = '{}/commits?sha=dev'.format(base_url)
+commit_pr_url = '{}/commits/commit_id/pulls'.format(base_url)
+
+history_line_breaker = '==============='
+days_before = 45
+history_notes = {}
```

Comment:
Only 45 days before?

Reply:
The commits on dev branch are all squashed commits generated by PR merge. So 45 days should be enough even consider cases like code freeze.

Reply:
Can we make it more preciously? Find the commit range between dev and release branch?

Reply:
I'm using this API: https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository. No support for parameter like starting commit.  

We can improve this later. For now as along as we don't hold release for more than 45 days. We won't miss all the commits generated by PR merge.

Reply:
git diff dev release

Reply:
updated the latest code to use git log range